### PR TITLE
feat: add extra plugin

### PR DESCRIPTION
**Pull request by Henrrypg**
This PR adds extra plugins extracted from https://github.com/eduNEXT/namespace-template/tree/main/template/%7B%25%20if%20tutor_version%20%3D%3D%20'olive'%20%25%7Denvironments%7B%25%20endif%20%25%7D/%7B%7Bnamespace%7D%7D/plugins

### CODE REVIEW

File: drydock/patches/kustomization [hunk 1]
```diff
@@ -46,3 +46,57 @@ patches:
...
+      path: /spec/template/spec/terminationGracePeriodSeconds
+      value: 300
+{% endif -%}
+{% if DRYDOCK_FORUM_OVERRIDES -%}
+- target:
+    kind: Deployment
+    name: forum
+  patch: |-
+    - op: add
+      path: /spec/template/spec/containers/0/env/-
```
Comment (jfavellar90) [inline drydock/patches/kustomization:96]: Do you need to keep the trailing hyphen for any specific reason?

**Comment by jfavellar90** [comment]
@Henrrypg I would like you to add the following [patch](https://github.com/eduNEXT/manifests-aeolus/blob/main/environments/cagliari/plugins/openedx-graceful-uwsgi.yml) as well. This patch adds the startup probes and grace periods for LMS and CMS uwsgi services. It should be toggleable.

File: drydock/patches/kustomization [hunk 2]
```diff
@@ -46,3 +46,109 @@ patches:
...
+      value:
+      - celery
+      - --app=cms.celery
+      - worker
+      - --loglevel=info
+      - --concurrency=1
+      - --hostname=edx.cms.core.default.%%h
+      - --max-tasks-per-child=1
+      - --prefetch-multiplier=1
+      - --exclude-queues=edx.lms.core.default
```
Comment (MoisesGSalas) [inline drydock/patches/kustomization:65]: I think these are better as straticmerge patches? We could them in a different files.

File: drydock/patches/kustomization [hunk 3]
```diff
@@ -46,3 +46,60 @@ patches:
...
+- path: plugins/drydock/k8s/celery-cms.yml
+- path: plugins/drydock/k8s/forum-overrides.yml
+{% endif -%}
+{% if DRYDOCK_GRACEFUL_UWSGI -%}
+- target:
+    kind: Deployment
+    name: lms|cms
+  patch: |-
+    - op: add
+      path: /spec/template/spec/containers/0/lifecycle
```
Comment (MoisesGSalas) [inline drydock/patches/kustomization:60]: The same comment applies to these patches, we can consolidate them in a single file.

File: drydock/templates/drydock/k8s/celery-cms.yml
[the remark this answers is not in the document]
Comment (Henrrypg) [reply drydock/templates/drydock/k8s/celery-cms.yml:1]: @MoisesGSalas  With `patch:` we cannot put more than one patch per file, i can move it to kustomization-patches-strategic-merge[](https://docs.tutor.overhang.io/reference/patches.html#kustomization-patches-strategic-merge) but it'll be deprecated

[the remark this answers is not in the document]
Comment (MoisesGSalas) [reply drydock/templates/drydock/k8s/celery-cms.yml:1]: Can we use different documents? separating them by `---`

[the remark this answers is not in the document]
Comment (Henrrypg) [reply drydock/templates/drydock/k8s/celery-cms.yml:1]: Nope, is not possible with patches ,throws the following error:

```
Error: trouble configuring builtin PatchTransformer with config: `
path: plugins/drydock/k8s/celery-lms.yml
`: unable to parse SM or JSON patch from [apiVersion: apps/v1
kind: Deployment
metadata:
  name: lms
spec:
  template:
    spec:
      containers:
        - name: lms
          lifecycle:
            preStop:
              exec:
                command: ["sleep", "30"]
          startupProbe:
            httpGet:
              httpHeaders:
                - name: Host
                  value: testdrydock-prod.atlas.edunext.link
              path: /heartbeat
              port: 8000
            initialDelaySeconds: 5
            timeoutSeconds: 3
            periodSeconds: 5
            failureThreshold: 5
      terminationGracePeriodSeconds: 60
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: cms
spec:
  template:
    spec:
      containers:
        - name: cms
          lifecycle:
            preStop:
              exec:
                command: ["sleep", "30"]
          startupProbe:
            httpGet:
              httpHeaders:
                - name: Host
                  value: studio.testdrydock-prod.atlas.edunext.link
              path: /heartbeat
              port: 8000
            initialDelaySeconds: 5
            timeoutSeconds: 3
            periodSeconds: 5
            failureThreshold: 5
      terminationGracePeriodSeconds: 60]
```
And looking the repo:
https://github.com/kubernetes-sigs/kustomize/issues/5052#issuecomment-1441421375

[the remark this answers is not in the document]
Comment (MoisesGSalas) [reply drydock/templates/drydock/k8s/celery-cms.yml:1]: That's a shame. Then I would suggest that we use different directories (e.g. `celery/`). 

BTW, the contents of the patch don't seem related to celery right? it should be named differently.

File: drydock/patches/cms-env [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{% if DRYDOCK_ENABLE_CELERY_TUNING -%}
+CELERY_ACKS_LATE: true
+{% endif %}
+{% if DRYDOCK_ENABLE_SENTRY -%}
+# eox plugin settings must be defined in the ENV patches. If defined in python settings patches, they won't
+# take effect since plugins settings are loaded before Tutor loads its settings.
+EOX_CORE_SENTRY_INTEGRATION_DSN: "{{ DRYDOCK_SENTRY_DSN }}"
+EOX_CORE_SENTRY_IGNORED_ERRORS: [
+    {
```
Comment (MoisesGSalas) [inline drydock/patches/cms-env:9]:
```suggestion
    {
      "exc_class": "xmodule.exceptions.NotFoundError"
    },
    {
```

**Review by MoisesGSalas — APPROVED** [review]

File: README.rst [hunk 5]
```diff
@@ -12,6 +12,14 @@ Drydock is an opinionated tool offering a set of Tutor plugins aiming to provide
 - Add a custom nginx and cert-manager configuration
 - Add a set of debug resources to help diagnose issues
 
+Extra plugins added:
+
+- A patch that allows for the tuning of celery workers via `DRYDOCK_ENABLE_CELERY_TUNING`
+- Allows caddy to catch requests for multiple domains through `DRYDOCK_ENABLE_MULTITENANCY`
+- Add scorm matcher to caddy through `DRYDOCK_ENABLE_SCORM`
+- A patch that allows for the use of sentry via `DRYDOCK_ENABLE_SENTRY` and `DRYDOCK_SENTRY_DSN`
+- Patch for cms and lms worker pods to allow pod probes, lifecycle and readiness probes to work properly. Enabled via `DRYDOCK_POD_LIFECYCLE`
```
Comment (jfavellar90) [inline README.rst:21]: We don't implement readiness probes, only startup probes

File: drydock/plugin.py [hunk 6]
```diff
@@ -28,6 +28,13 @@
         "NEWRELIC_LICENSE_KEY": "",
         "CUSTOM_CERTS": {},
         "DEBUG": False,
+        "ENABLE_CELERY_TUNING": True,
+        "ENABLE_MULTITENANCY": True,
+        "ENABLE_SCORM": True,
+        "ENABLE_SENTRY": True,
+        "SENTRY_DSN": "",
+        "GRACEFUL_UWSGI": False,
```
Comment (jfavellar90) [inline drydock/plugin.py:36]: Is this one used?

**Review by jfavellar90 — APPROVED** [review]

**Pull request merged**